app/models/competition.rb

```diff
@@ -1767,6 +1767,45 @@ def self.search(query, params: {}, managed_by_user: nil)
     competitions.includes(:delegates, :organizers, *previous_includes).order(**order)
   end
 
+  def competing_step_parameters(current_user)
+    competition_params = serializable_hash(only: [:events_per_registration_limit,
+                             :allow_registration_edits,
+                             :guest_entry_status,
+                             :guests_per_registration_limit,
+                             :guests_enabled,
+                             :uses_qualification?,
+                             :allow_registration_without_qualification,
+                             :force_comment_in_registration],
+                      methods: [:qualification_wcif, :event_ids],
+                      include: [])
+    user_params = {
+      preferredEvents: current_user.preferred_events.pluck(:id),
+      personalRecords: {
+        single: current_user.ranks_single&.map(&:to_wcif) || [],
+        average: current_user.ranks_average&.map(&:to_wcif) || [],
+      },
+    }
+    competition_params.merge!(user_params)
```

Comment:
`merge` (without the exclamation point) should be sufficient here. `merge!` modifies `competition_params` in-place, but since you're caring only about the overall return value, the result of `merge` should be enough.